ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -363,6 +363,33 @@ module Vendor_nvidia = struct
   let pt_when_vgpu = true
   let whitelist_file () = !Xapi_globs.nvidia_whitelist
   let device_id_of_conf conf = conf.identifier.Identifier.pdev_id
+  
+  let get_host_driver_version () = try Scanf.sscanf (Unix.readlink !Xapi_globs.nvidia_host_driver_file) "libnvidia-vgpu.so.%s" (fun x -> x)
+    with _ -> info "Can not get the file version of %s, use the default value" !Xapi_globs.nvidia_host_driver_file; Xapi_globs.nvidia_default_host_driver_version
+
+  let is_host_driver_support_multi_vgpu ~host_driver_version ~supported_versions =
+    List.mem host_driver_version supported_versions ||
+    (
+      let valid_version_pattern = Re.(compile @@ Pcre.re {|^([0-9]+)(\+?)\.([0-9]+)(\+?)$|}) in
+      (* 0. total string, 1. major version, 2. + or "" 3. minor version. 4. + or "" *)
+      let host_driver_major_version, host_driver_minor_version =
+        try Re.Pcre.extract valid_version_pattern host_driver_version |> (fun x -> Array.get x 1 |> int_of_string, Array.get x 3 |> int_of_string)
+        with Not_found -> raise (Api_errors.Server_error (Api_errors.internal_error,[Printf.sprintf "Failed to parse host driver version: %s" host_driver_version])) in
+      List.filter_map (fun x -> try Some (Re.Pcre.extract valid_version_pattern x) with _ -> None) supported_versions
+      |> List.filter (fun x -> Array.get x 2 = "+" || Array.get x 4 = "+") (* Filter all version that from them on *)
```

Comment:
This `filter` can only be passed by a version that has a `+` in it. Why does this pass the unit tests that have no `+`?

```
        ("430.19",["430.89";"430.19"]), true; (* Test specific version should match *)
```

Reply:
see ` List.mem host_driver_version supported_versions`, but never mind, I will change the function as Rob suggested.